Tests/SwiftGRPCNIOTests/NIOServerTests.swift

```diff
@@ -92,211 +94,106 @@ class NIOServerTests: NIOServerTestCase {
     ]
   }
 
-  static let lotsOfStrings = (0..<1000).map { String(describing: $0) }
-
-  var eventLoopGroup: MultiThreadedEventLoopGroup!
-  var server: GRPCServer!
-
-  override func setUp() {
-    super.setUp()
-
-    // This is how a GRPC server would actually be set up.
-    eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-    server = try! GRPCServer.start(
-      hostname: "localhost", port: 5050, eventLoopGroup: eventLoopGroup, serviceProviders: [EchoProvider_NIO()])
-      .wait()
-  }
-
-  override func tearDown() {
-    XCTAssertNoThrow(try server.close().wait())
-
-    XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
-    eventLoopGroup = nil
-
-    super.tearDown()
-  }
+  static let aFewStrings = ["foo", "bar", "baz"]
+  static let lotsOfStrings = (0..<10_000).map { String(describing: $0) }
 }
 
 extension NIOServerTests {
-  func testUnary() {
-    XCTAssertEqual("Swift echo get: foo", try! client.get(Echo_EchoRequest(text: "foo")).text)
+  func testUnary() throws {
+    let options = CallOptions(timeout: nil)
+    XCTAssertEqual(try client.get(Echo_EchoRequest.with { $0.text = "foo" }, callOptions: options).response.wait().text, "Swift echo get: foo")
   }
 
-  func testUnaryLotsOfRequests() {
+  func testUnaryLotsOfRequests() throws {
     // Sending that many requests at once can sometimes trip things up, it seems.
-    client.timeout = 5.0
     let clockStart = clock()
     let numberOfRequests = 2_000
+
     for i in 0..<numberOfRequests {
       if i % 1_000 == 0 && i > 0 {
         print("\(i) requests sent so far, elapsed time: \(Double(clock() - clockStart) / Double(CLOCKS_PER_SEC))")
       }
-      XCTAssertEqual("Swift echo get: foo \(i)", try client.get(Echo_EchoRequest(text: "foo \(i)")).text)
+      XCTAssertEqual(try client.get(Echo_EchoRequest.with { $0.text = "foo \(i)" }).response.wait().text, "Swift echo get: foo \(i)")
     }
     print("total time for \(numberOfRequests) requests: \(Double(clock() - clockStart) / Double(CLOCKS_PER_SEC))")
   }
 }
 
 extension NIOServerTests {
-  func testClientStreaming() {
-    let completionHandlerExpectation = expectation(description: "final completion handler called")
-    let call = try! client.collect { callResult in
-      XCTAssertEqual(.ok, callResult.statusCode)
-      completionHandlerExpectation.fulfill()
-    }
-
-    XCTAssertNoThrow(try call.send(Echo_EchoRequest(text: "foo")))
-    XCTAssertNoThrow(try call.send(Echo_EchoRequest(text: "bar")))
-    XCTAssertNoThrow(try call.send(Echo_EchoRequest(text: "baz")))
-    call.waitForSendOperationsToFinish()
+  func doTestClientStreaming(messages: [String]) throws {
```

Comment:
I'd add `file: StaticString = #file, line: Int = #line` to all method signatures like this and passing those to the asserts; that makes it easier to pinpoint which test failed.